src/devtools/client/debugger/src/actions/breakpoints/modify.js

```diff
@@ -137,9 +138,21 @@ export function addBreakpoint(
     if (disabled) {
       // If we just clobbered an enabled breakpoint with a disabled one, we need
       // to remove any installed breakpoint in the server.
-      await client.removeBreakpoint(location);
-    } else {
-      await client.setBreakpoint(breakpoint.location, breakpoint.options);
+      return await client.removeBreakpoint(location);
+    }
+
+    const [oldLogGroupId, logGroupId] = await client.setBreakpoint(
+      breakpoint.location,
+      breakpoint.options
+    );
+
+    // Keep a reference to the analysis's log group id in redux. That way, we can reference
+    // and clear it if we happen to remove the breakpoint before the analysis is finished.
+    dispatch(setBreakpointOptions(cx, location, { ...options, logGroupId }));
+
+    // This is defensive just to make sure we always clear unneeded logGroupIds.
+    if (oldLogGroupId !== logGroupId) {
+      dispatch(removeLogGroupId(oldLogGroupId));
```

Comment:
can we do this all in the reducer? feels like this could be simpler. For instance, the console reducer could listen for the `setBreakpointOptions` action and see ooh, we're trying to change an already existing analysis... also - another option is to use a `location-key` which might be simpler than log-group-id.